src/Fixer/FunctionNotation/NoTrailingCommaInSinglelineFunctionCallFixer.php

```diff
@@ -0,0 +1,112 @@
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'When making a method or function call on a single line there MUST NOT be a trailing comma after the last argument.',
+            [new CodeSample("<?php\nfoo(\$a,);\n")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before NoSpacesInsideParenthesisFixer.
+     */
+    public function getPriority(): int
+    {
+        return 3;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound([T_STRING, T_VARIABLE, T_CLASS, T_UNSET, T_ISSET, T_LIST]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = \count($tokens) - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->equals(')')) {
+                continue;
+            }
+
+            $trailingCommaIndex = $tokens->getPrevMeaningfulToken($index);
+
+            if (!$tokens[$trailingCommaIndex]->equals(',')) {
+                continue;
+            }
+
+            $callIndex = $tokens->getPrevMeaningfulToken( // get before "parenthesis open index"
+                $tokens->findBlockStart(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index)
+            );
+
+            if ($tokens[$callIndex]->isGivenKind([T_VARIABLE, T_CLASS, T_UNSET, T_ISSET, T_LIST])) {
+                $this->clearCommaIfNeeded($tokens, $callIndex, $index, $trailingCommaIndex);
+
+                continue;
+            }
+
+            if ($tokens[$callIndex]->isGivenKind(T_STRING)) {
+                if (!AttributeAnalyzer::isAttribute($tokens, $callIndex)) {
+                    $this->clearCommaIfNeeded($tokens, $callIndex, $index, $trailingCommaIndex);
+                }
+
+                continue;
+            }
+
+            if ($tokens[$callIndex]->equalsAny([')', ']', [CT::T_DYNAMIC_VAR_BRACE_CLOSE], [CT::T_ARRAY_INDEX_CURLY_BRACE_CLOSE]])) {
+                $block = Tokens::detectBlockType($tokens[$callIndex]);
+                if (
+                    Tokens::BLOCK_TYPE_ARRAY_INDEX_CURLY_BRACE === $block['type']
+                    || Tokens::BLOCK_TYPE_DYNAMIC_VAR_BRACE === $block['type']
+                    || Tokens::BLOCK_TYPE_INDEX_SQUARE_BRACE === $block['type']
+                    || Tokens::BLOCK_TYPE_PARENTHESIS_BRACE === $block['type']
+                ) {
+                    $this->clearCommaIfNeeded($tokens, $callIndex, $index, $trailingCommaIndex);
+
+                    // continue; implicit
+                }
+            }
+        }
+    }
+
+    private function clearCommaIfNeeded(Tokens $tokens, int $startIndex, int $endIndex, int $commaIndex): void
+    {
+        if (!$tokens->isPartialCodeMultiline($startIndex, $endIndex)) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($commaIndex);
```

Comment:
Why this way, not:
```php
            $tokens->removeTrailingWhitespace($beforeEndIndex);
            $tokens->clearAt($beforeEndIndex);
```
as it's done in `NoTrailingCommaInSinglelineArrayFixer`?

Reply:
not sure I follow, what would be the benefit?

Reply:
Consistency, better fixing - now code:
```php
$a = [1, 2, 3, ];
foo(1, 2, 3, );
```
would be changed to:
```php
$a = [1, 2, 3];
foo(1, 2, 3 );
```

I cannot imagine anyone want to have the comma removed and keep it's trailing whitespace.

Reply:
check, I didn't even know `removeTrailingWhitespace` existed, I think your case is covered through single-responsibility of the rule combined with priorities, but if there is more room for improvement than lets PR it.